Sming/SmingCore/Rtc.cpp

```diff
@@ -0,0 +1,66 @@
+#include "../SmingCore/Rtc.h"
+
+RTC::RTC() {
+	rst_info* info = system_get_rst_info();
+	hardwareReset = (info->reason == REASON_WDT_RST);
+}
+
+uint32_t RTC::getRtcSeconds() {
+	RtcData rtcTime;
+	readFromRtcMemory(rtcTime);
+	updateRtcTime(rtcTime);
+	saveToRtcMemory(rtcTime);
+	return (rtcTime.time / NS_PER_SECOND);
+}
+
+
+
+bool RTC::setRtcSeconds(uint32_t seconds) {
+
+	RtcData rtcTime;
+	readFromRtcMemory(rtcTime);
+	uint64_t offsetNs = ((uint64_t) seconds * NS_PER_SECOND) - rtcTime.time;
+	rtcTime.time += offsetNs; // adjust time in nanoseconds by the ntp time offset in nanoseconds
+	updateRtcTime(rtcTime);
+	return saveToRtcMemory(rtcTime);
+}
+
+
+void RTC::updateRtcTime(RtcData &data) {
+
+	uint32 rtc_cycles;
+	uint32 cal, cal1, cal2;
+	cal1 = system_rtc_clock_cali_proc();
+	os_delay_us(300);
+	cal2 = system_rtc_clock_cali_proc();
+	cal = (cal1 + cal2) / 2; // get average cal in case one is out

```

Comment:
Sorry, I have misunderstanding system_rtc_clock_cali_proc method functionality, but still want to know why addiotional delay is needed here? To be more accurate? What if updateRtcTime will be called to often (from getRtcSeconds at first)?